src/main/java/com/cedrus/patpetillo/springkafkapingpong/stream/TopologyProvider.java

```diff
@@ -0,0 +1,134 @@
+package com.cedrus.patpetillo.springkafkapingpong.stream;
+
+import com.cedrus.patpetillo.springkafkapingpong.config.AppConfig;
+import com.cedrus.patpetillo.springkafkapingpong.config.TopicConfig;
+import com.cedrus.patpetillo.springkafkapingpong.model.PingPongBall;
+import com.cedrus.patpetillo.springkafkapingpong.model.PingPongTarget;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.kstream.*;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+@Slf4j
+@Component
+public class TopologyProvider {
+
+    private final TopicConfig topicConfig;
+    private final AppConfig appConfig;
+    private final ObjectMapper objectMapper;
+
+    @Autowired
+    public TopologyProvider(TopicConfig topicConfig, AppConfig appConfig, ObjectMapper objectMapper){
+        this.topicConfig = topicConfig;
+        this.appConfig = appConfig;
+        this.objectMapper = objectMapper;
+    }
+
+    public Topology getTopology(PingPongTarget pingPongTarget) {
+        final StreamsBuilder builder = new StreamsBuilder();
+        log.info("Stream builder initialized");
+        log.info(pingPongTarget.toString());
+
+        final KStream<String, String> incomingStream = builder.stream(topicConfig.getTopicName(), Consumed.with(Serdes.String(), Serdes.String()));
+
+        final KStream<String, String> pingPongStream = incomingStream.filter((key, value) -> value.contains("pingPongTarget"));
+
+        final KStream<String, String> filteredStream = pingPongStream.filter((key, value) -> {
+            PingPongBall pingPongBall = deserialize(value);
+            log.info("deserialize value: {}", pingPongBall);
+            return pingPongBall.getPingPongTarget().equals(pingPongTarget);
+        });
+
+        final KStream<String, String> volleyCount = filteredStream
+                .groupByKey()
+                .count()
+                .mapValues((key, value) -> {
```

Comment:
This is a styling decision, and you're not wrong to have done this, but it breaks up your Streams structure with processing instructions.  Consider putting this into a private method and having that return your functional object.  It really cleans things up.  Reading this, I don't actually know what class that is.  You can put the lambda expression into the private method.